src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -912,6 +912,44 @@ private void SpillEvalStack()
             }
         }
 
+        private void PushArray<T>(ImmutableArray<T> array, Func<T, IOperation> unwrapper = null) where T : IOperation
+        {
+            Debug.Assert(unwrapper != null || typeof(T) == typeof(IOperation));
+            foreach (var element in array)
+            {
+                _evalStack.Push(Visit(unwrapper == null ? element : unwrapper(element)));
+            }
+        }
+
+        private ImmutableArray<T> PopArray<T>(ImmutableArray<T> originalArray, Func<IOperation, int, ImmutableArray<T>, T> wrapper = null) where T : IOperation
+        {
+            Debug.Assert(wrapper != null || typeof(T) == typeof(IOperation));
+            int numElements = originalArray.Length;
+            if (numElements == 0)
+            {
+                return ImmutableArray<T>.Empty;
+            }
+            else
+            {
+                var builder = ArrayBuilder<T>.GetInstance(numElements);
+                // Iterate in reverse order so the index corresponds to the original index when pushed onto the stack
+                for (int i = numElements - 1; i >= 0; i--)
+                {
+                    IOperation visitedElement = _evalStack.Pop();
+                    if (wrapper != null)
+                    {
+                        builder.Add(wrapper(visitedElement, i, originalArray));
+                    }
+                    else
+                    {
+                        builder.Add((T)visitedElement);
+                    }
```

Comment:
Consider using `?:` for argument to `builder.Add()` rather than `if (wrapper != null)`.